### PR TITLE
feat: add browser esm build

### PR DESCRIPTION
**Pull request by antfu**
Follow up #109 

- Inlined `join` and `dirname` so it no longer relays on `path` in browser build.
- New ESM build for browser under `./dist/index.browser.mjs` where you would need to import explicitly with full URL when using in the browser.

/cc @43081j

### CODE REVIEW

**Comment by antfu** [comment]
Tests failed due to the missing of `dark-vs.json` and `light-vs.json`, which I supposed is not introduced in this PR. Do you have any clue about how would that happen? (also seems like the tests are not triggered for the commits on the master branch)

**Comment by antfu** [comment]
Ah, should be this one https://github.com/shikijs/shiki/commit/840a6cf82a88038e28babc1043e3d28f3a32eb0c

File: packages/shiki/rollup.config.js [hunk 1]
```diff
@@ -35,6 +35,17 @@ export default [
   {
     input: 'src/index.ts',
     output: [
+      {
+        file: 'dist/index.browser.mjs',
+        format: 'esm',
+        name: 'shiki',
+        extend: true,
```
[the remark this answers is not in the document]
Comment (antfu) [reply packages/shiki/rollup.config.js]: Just copied from iife build without a second thought, thanks for catching it.

**Comment by 43081j** [comment]
looks good, i tried it in chrome and it works well.

just so i understand, what is it that bundles the dependencies into the output file? its right, i just wondered where that happens as the two rollup bundles look identical except the browser one has `terser`, so is it terser which causes it to be one single-file bundle?

**Comment by antfu** [comment]
@43081j the difference is that the `external` options which control whether the deps should be bundled. `terser` is used for minification.

**Comment by 43081j** [comment]
yeah i thought so, thats why i was confused :D i didn't notice the `external` part! thanks for pointing that out

**Pull request merged**